pkg/resourcemanager/controller/managedresource/merger.go

```diff
@@ -146,16 +147,19 @@ func mergeDeployment(scheme *runtime.Scheme, oldObj, newObj runtime.Object, pres
 }
 
 func mergePodTemplate(oldPod, newPod *corev1.PodTemplateSpec, preserveResources bool) {
-	if !preserveResources {
-		return
-	}
-
-	// Do not overwrite a PodTemplate's resource requests / limits if it is scaled by an HVPA
-	for i, newContainer := range newPod.Spec.Containers {
-		for j, oldContainer := range oldPod.Spec.Containers {
-			if newContainer.Name == oldContainer.Name {
-				mergeContainer(&oldPod.Spec.Containers[j], &newPod.Spec.Containers[i], preserveResources)
-				break
+	// ref https://kubernetes.io/docs/reference/labels-annotations-taints/#kubectl-k8s-io-restart-at
```

Comment:
You could add a more comprehensive comment on why we need to preserve this annotation:

```suggestion
	// Do not overwrite the "kubectl.kubernetes.io/restartedAt" annotation as it is used by
	// the "kubectl rollout restart <RESOURCE>" command to trigger rollouts.
	// Otherwise, the resource-manager would revert this annotation set by the kubectl command and
	// this would revert the triggered rollout.
	// Ref https://kubernetes.io/docs/reference/labels-annotations-taints/#kubectl-k8s-io-restart-at
```